### PR TITLE
Migration Guide: 0.x to 1.0

### PR DESCRIPTION
**Pull request by suhsteve**
Add `Microsoft.Spark` 0.x to 1.0 migration guide.

### CODE REVIEW

File: docs/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Migration Guide
+- [Upgrading from Microsoft.Spark 0.x to 1.0](#upgrading-from-microsoft.spark-0.x-to-1.0)
+
+## Upgrading from Microsoft.Spark 0.x to 1.0
+- `Microsoft.Spark.Worker` should be upgraded to 1.0 as `Microsoft.Spark.Worker` 0.x is not foward-compatible with `Microsoft.Spark` 1.0.
```
Comment (Niharikadutta) [inline docs/migration-guide.md:5]:
```suggestion
- `Microsoft.Spark.Worker` should be upgraded to 1.0 as `Microsoft.Spark.Worker` 0.x is not forward-compatible with `Microsoft.Spark` 1.0.
```

**Comment by Niharikadutta** [comment]
Are we not mentioning the following:

- [ ] Known issues: e.g.) grouped map UDF for Spark 3.0
- [ ] .NET Delta/Hyperspace version | OSS Delta/Hyperspace version | Spark versions matrix
- [ ] Dropping support for .NET Framework

**Comment by Niharikadutta** [comment]
@imback82 Are planning on adding scala version to the `microsoft-spark*.jar` filename?

**Comment by suhsteve** [comment]
> Are we not mentioning the following:
> 
> * [ ]  Known issues: e.g.) grouped map UDF for Spark 3.0
> * [ ]  .NET Delta/Hyperspace version | OSS Delta/Hyperspace version | Spark versions matrix
> * [ ]  Dropping support for .NET Framework

The grouped map for 3.0 should go to the Release notes.  Limited support for .NET Framework maybe the release notes as well ?   Don't know if we need a delta/hyperspace matrix or not and where that should go.

**Comment by suhsteve** [comment]
> @imback82 Are we planning on adding scala version to the `microsoft-spark*.jar` filename?

we can update release notes and this guide once the name is finalized.

File: docs/migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Migration Guide
+- [Upgrading from Microsoft.Spark 0.x to 1.0](#upgrading-from-microsoft.spark-0.x-to-1.0)
```
Comment (Niharikadutta) [inline docs/migration-guide.md:2]: Do we need this to be a self-referencing link?

Comment (Niharikadutta) [reply docs/migration-guide.md:2]: Or is it there coz we might add more sections to this doc

Comment (suhsteve) [reply docs/migration-guide.md:2]: We should update this file every time we need new migration guide steps between versions.  The table of contents is there so we can add more sections later.

File: docs/migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Migration Guide
+- [Upgrading from Microsoft.Spark 0.x to 1.0](#upgrading-from-microsoftspark-0x-to-10)
+
+## Upgrading from Microsoft.Spark 0.x to 1.0
+- Limited support for [.NET Framework](https://dotnet.microsoft.com/learn/dotnet/what-is-dotnet-framework). Please migrate to [.NET Core >= 3.1](https://dotnet.microsoft.com/download/dotnet-core) instead.
```
Comment (imback82) [inline docs/migration-guide.md:5]: Can you mention that `ForEach` doesn't work (and link the existing issue)?

File: docs/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+- `Microsoft.Spark.Worker` should be upgraded to 1.0 as `Microsoft.Spark.Worker` 0.x is not forward-compatible with `Microsoft.Spark` 1.0.
+- `Microsoft.Spark` should be upgraded to 1.0 as `Microsoft.Spark.Worker` 1.0 is not backward-compatible with `Microsoft.Spark` 0.x.
+- `Microsoft.Spark.Experimental` project has been merged into `Microsoft.Spark`
+  - `VectorUdf` from `Microsoft.Spark.Sql.ExperimentalFunctions` is now part of `Microsoft.Spark.Sql.ArrowFunctions`.
+  - `VectorUdf` from `Microsoft.Spark.Sql.ExperimentalDataFrameFunctions` is now part of `Microsoft.Spark.Sql.DataFrameFunctions`.
+  - Extension methods have been moved from `Microsoft.Spark.Sql.RelationalGroupedDatasetExtensions` into the main `Microsoft.Spark.Sql.RelationalGroupedDataset` class.
```
Comment (imback82) [inline docs/migration-guide.md:11]: I guess we can remove main?

**Review by imback82 — DISMISSED** [review]
LGTM except few minor comments, thanks @suhsteve!

**Comment by imback82** [comment]
> > @imback82 Are we planning on adding scala version to the `microsoft-spark*.jar` filename?
> 
> we can update release notes and this guide once the name is finalized.

Sorry for the delay. I will sort this out tomorrow.

**Comment by imback82** [comment]
> Don't know if we need a delta/hyperspace matrix or not and where that should go.

Let's start putting this in the release notes similar to Spark versions supported.

**Comment by imback82** [comment]
The naming for the JAR is finalized: #293. Can @suhsteve / @Niharikadutta help with updating any docs? Thanks!

**Comment by suhsteve** [comment]
> The naming for the JAR is finalized: #293. Can @suhsteve / @Niharikadutta help with updating any docs? Thanks!

I've updated the docs in the repo.  #729

File: docs/migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Migration Guide
+- [Upgrading from Microsoft.Spark 0.x to 1.0](#upgrading-from-microsoftspark-0x-to-10)
+
+## Upgrading from Microsoft.Spark 0.x to 1.0
+- Jar name has changed.  `microsoft-spark-2.3.x-<version>.jar`,`microsoft-spark-2.4.x-<version>.jar`,`microsoft-spark-3.0.x-<version>.jar` have been renamed to `microsoft-spark-2-3_2.11-1.0.0.jar`, `microsoft-spark-2-4_2.11-1.0.0.jar`, `microsoft-spark-3-0_2.12-1.0.0.jar` respectively. ([#293](https://github.com/dotnet/spark/issues/293))([#728](https://github.com/dotnet/spark/issues/728))
```
Comment (imback82) [inline docs/migration-guide.md:5]: How about using some table?

Old JAR | New JAR
---------|---------
microsoft-spark-2.3.x-<version>.jar | microsoft-spark-2-3_2.11-1.0.0.jar
microsoft-spark-2.4.x-<version>.jar | microsoft-spark-2-4_2.11-1.0.0.jar
microsoft-spark-3.0.x-<version>.jar | microsoft-spark-3-0_2.12-1.0.0.jar

Comment (suhsteve) [reply docs/migration-guide.md]: updated.

**Review by imback82 — APPROVED** [review]
LGTM, thanks @suhsteve!

**Pull request merged**